### PR TITLE
Added the option to remove (multiple) saved queries and queries from the history

### LINKED ISSUE #305: FEAT: Button to clear history of queries or saved queries
An option that allows you to delete queries you no longer need, either individually or as a group

### PR DESCRIPTION
**Pull request by UtechtDustin**
Added the option to remove (multiple) saved queries and queries from the history

Fix #305

Peek 2020-08-30 01-52

### CODE REVIEW

**Comment by UtechtDustin** [comment]
@rathboma 
It's working fine for the saved queries, so i tried to implement the same for the quries in the history but i get the error
```
 Property or method "discardCheckedHistoryQueries" is not defined on the instance but referenced during render. Make sure that this property is reactive, either in the data option, or for class-based components, by initializing the property.
```
I dont understand that error, because there is a method called `discardCheckedHistoryQueries`, also its working for the saved queries.
Could you help with that, did i miss something ?!

**Comment by rathboma** [comment]
Sure I'll take a look. @ghstng can you do a ui review?

**Comment by ghstng** [comment]
I can pull into a branch and cleanup, if you both are good with everything.

**Comment by UtechtDustin** [comment]
Its fine for me, im not a UI expert :-)

**Comment by rathboma** [comment]
Sure then I can fix the vue problem

**Comment by UtechtDustin** [comment]
> Sure then I can fix the vue problem

Thanks, please then tell me what was the issue was :-)

**Comment by rathboma** [comment]
@UtechtDustin Your bug is that `discardCheckedHistoryQueries` is outside of the `methods {}` block, simply move it inside `methods`, and it will work (also do that for the other function you added)

**Comment by UtechtDustin** [comment]
> @UtechtDustin Your bug is that `discardCheckedHistoryQueries` is outside of the `methods {}` block, simply move it inside `methods`, and it will work (also do that for the other function you added)

Oh... How i can't saw that.. Sometimes the small issues are the hardest one..

**Comment by rathboma** [comment]
If you fix that little bug we'll merge it and do a style tidy up 👍

File: src/components/sidebar/core/FavoriteList.vue [hunk 1]
```diff
@@ -28,15 +35,32 @@
...
     methods: {
       click(item) {
         this.$root.$emit('favoriteClick', item)
       },
       async remove(favorite) {
         await this.$store.dispatch('removeFavorite', favorite)
+      },
+      async removeCheckedFavorites() {
+        this.checkedFavorites.forEach(favorite => {
+          this.remove(favorite)
```
Comment (rathboma) [inline src/components/sidebar/core/FavoriteList.vue:58]: Should you have an `await` here? You will need to do a for() loop to do that

File: src/components/sidebar/core/HistoryList.vue [hunk 2]
```diff
@@ -35,7 +57,18 @@
           return text
         }
       }
+    },
+    async remove(historyQuery) {
+      await this.$store.dispatch('removeHistoryQuery', historyQuery)
+    },
+    async removeCheckedHistoryQueries() {
+      this.checkedHistoryQueries.forEach(historyQuery => {
+        this.remove(historyQuery)
```
Comment (rathboma) [inline src/components/sidebar/core/HistoryList.vue:66]: same deal with `await` here

**Review by rathboma — CHANGES_REQUESTED** [review]
minor change - do an `await` wherever you can

**Comment by UtechtDustin** [comment]
Lets wait for @ghstng for the design then i will look at this :-)

**Comment by rathboma** [comment]
We chatted -- the ux is mostly good, we're just going to tweak a few things after it's merged -- it's hard to do that with someone else's branch

**Comment by UtechtDustin** [comment]
Oh okay, i pushed a new version of that.
Currently i have one last problem (except from the design), if i remove saved query somehow ALL queries will be removed from the list but no one will be removed from the store.
I try to figure out what the problem is, but if you have time it would great if you also could look.

**Comment by UtechtDustin** [comment]
@rathboma its ready for merge

**Pull request merged**